tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1458,11 +1458,24 @@ void CorProfiler::AddDerivedInstrumentations(WCHAR* id, CallTargetDefinition* it
 
     if (size > 0)
     {
-        InternalAddInstrumentation(id, items, size, true);
+        InternalAddInstrumentation(id, items, size, true, false);
+    }
+}
+
+void CorProfiler::AddInterfaceInstrumentations(WCHAR* id, CallTargetDefinition* items, int size)
+{
+    auto _ = trace::Stats::Instance()->InitializeProfilerMeasure();
+    shared::WSTRING definitionsId = shared::WSTRING(id);
+    Logger::Info("AddInterfaceInstrumentations: received id: ", definitionsId, " from managed side with ", size,
+                 " integrations.");
+
+    if (size > 0)
+    {
+        InternalAddInstrumentation(id, items, size, false, true);
     }
 }
 
-void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* items, int size, bool isDerived, bool enable)
+void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* items, int size, bool isDerived, bool isInterface, bool enable)
```

Comment:
Maybe for other PR, but we should start thinking on moving the two booleans to an enum.